plan/plan.go

```diff
@@ -37,6 +37,8 @@ type Plan struct {
 	Current []*endpoint.Endpoint
 	// List of desired records
 	Desired []*endpoint.Endpoint
+	// List of missing records to be created, use for the migrations (e.g. old-new TXT format)
+	Missing []*endpoint.Endpoint
```

Comment:
Is there any other way that you see why we need to modify the Plan? Can we modify `generateTXTRecord` to add both the records instead? /cc @k0da

Reply:
@Raffo: `generateTXTRecord` is used in the registry after the plan is calculated. The plan contains things to be created/updated/deleted. The records created by the previous version of externaldns will not get they didn't change. So, the records whose source didn't change after the migration to `v0.12.x` will still have a single TXT record.

Talking about the modification of the plan: I was trying to choose between modifying `Plan` `struct` and putting my hands into the plan calculation (e.g. adding missing TXT records into the slice of desired endpoints). And I thought that less fitting but clear and explicit field in `struct` would be easier that modifying the cornerstone algorithm of the plan calculation.
`Plan` struct could have been modified with a new method instead of the field. That new method could add the missing records into the `create` changes. But would it be less intrusive than a field? I don't know, kinda the same to me.

Reply:
Another thing I considered is in the `Registry` interface: `Missing` method. I didn't want to introduce a new method which would be used only in the TXT registry and only for the migration (btw it's not unlikely that this is not our last migration :) ). First I thought of returning the missing TXT records in the same slice as the other records, but this slice has non missing TXT records too, so hard to distinguish. And again, as in the previous comment, I decided to be less "sexy" but simple and explicit.